state/cluster/badger/mutator.go

```diff
@@ -120,7 +120,14 @@ func (m *MutableState) Extend(block *cluster.Block) error {
 		checkTxsSpan, _ := m.tracer.StartSpanFromContext(ctx, trace.COLClusterStateMutatorExtendCheckTransactionsValid)
 		defer checkTxsSpan.Finish()
 
+		// no validation of transactions is necessary for empty collections
+		if payload.Collection.Len() == 0 {
+			return nil
```

Comment:
Good catch!

But we still need to validate `payload.ReferenceBlockID` exist

Reply:
I think it is actually fine to skip that for empty collections, since they are not sent to consensus nodes. We have an issue to define and check a specific value for empty collections' reference block ID: https://github.com/dapperlabs/flow-go/issues/6245. But since the field is never checked currently, I think we can leave that to be addressed with #6245